### PR TITLE
[Sparse] [llvm] Support memory profiling for LLVM backends

### PR DESCRIPTION
**Pull request by yuanming-hu**
Sample outputs:
```
[Memory Profiler]
SNode S0root    
  active element list: length=1     1 chunks x [65,536 x 48 B]  total=3.1457 MB
SNode S1dense   
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
SNode S3dense   
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
SNode S7dense   
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
SNode S11dense  
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
SNode S15pointer
  active element list: length=32     1 chunks x [65,536 x 48 B]  total=3.1457 MB
  data list:           length=7     1 chunks x [2,048 x 65,536 B]  total=134.2177 MB
  Allocated elements=7; free list length=7; recycled list length=0
SNode S16pointer
  active element list: length=28     1 chunks x [65,536 x 48 B]  total=3.1457 MB
  data list:           length=2,705     1 chunks x [8,192 x 8,208 B]  total=67.2399 MB
  Allocated elements=2,663; free list length=2,705; recycled list length=0
SNode S17dense  
  active element list: length=2,663     1 chunks x [65,536 x 48 B]  total=3.1457 MB
SNode S19dense  
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
SNode S21dense  
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
SNode S23dense  
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
SNode S25dynamic
  active element list: length=1,401     1 chunks x [65,536 x 48 B]  total=3.1457 MB
  data list:           length=1,203     1 chunks x [4,096 x 16,392 B]  total=67.1416 MB
  Allocated elements=1,177; free list length=1,203; recycled list length=0
SNode S27dynamic
  active element list: length=612     1 chunks x [65,536 x 48 B]  total=3.1457 MB
  data list:           length=1     1 chunks x [1 x 113,246,216 B]  total=113.2462 MB
  Allocated elements=1; free list length=0; recycled list length=0
SNode S55dense  
  active element list: length=0     0 chunks x [65,536 x 48 B]  total=0.0000 MB
Total requested dynamic memory (excluding alignment padding): 514,739,152 B

```

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: taichi/backends/cuda/cuda_driver.h [hunk 1]
```diff
@@ -60,9 +60,10 @@ class CUDADriverFunction {
            fmt::format(" while calling {} ({})", name, symbol_name);
   }
 
-  void call_with_warning(Args... args) {
+  uint32 call_with_warning(Args... args) {
     auto err = call(args...);
     TI_WARN_IF(err, get_error_message(err));
```
Comment (archibate) [inline taichi/backends/cuda/cuda_driver.h:65]:
```suggestion
    TI_WARN_IF(err, "{}", get_error_message(err));
```

**Comment by archibate** [comment]
Hi, is this `memory profiler` only designed for backends with full sparse support as I read from the PR desc? Not sure if I could do something to support this on OpenGL, which doesn't have a mature list manager yet.

File: misc/prtags.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   "opengl"          : "OpenGL backend",
   "misc"            : "Miscellaneous",
   "std"             : "Standard library",
+  "sparse"          : "Sparse computation",
```
Comment (archibate) [inline misc/prtags.json:13]:
```suggestion
```
Please see line 18.

**Comment by yuanming-hu** [comment]
> Hi, is this `memory profiler` only designed for backends with full sparse support as I read from the PR desc? Not sure if I could do something to support this on OpenGL, which doesn't have a mature list manager yet.

Good question. Yes, it's for sparse computation only. I added a "sparse" tag for more clarity.

No rush adding that to OpenGL. It is only meaningful when you have full sparse computation support.

**Pull request merged**